doc/doxygen/src/creating-an-application.md

```diff
@@ -0,0 +1,92 @@
+Creating an application                              {#creating-an-application}
+=======================
+
+[TOC]
+
+To create your own application you need to create a directory contating some 
+from the C file(s) containing your source code and a Makefile. A template
+Makefile is available in the `dist` folder of the
+[RIOT repository](https://github.com/RIOT-OS/RIOT).
+
+The main function                                          {#the-main-function}
+=================
+After the board is initialized, RIOT starts two threads: the idle thread and 
+the main thread. The idle thread has the lowest priority and will run whenever 
+no other thread is ready to run. It will automatically use the lowest possible 
+power mode for the device. The main thread - configured with a default priority 
+that is right in the middle between the lowest and the highest available 
+priority - is the first thread that runs and calls the `main()` function. This 
+function needs to be defined in the source code of the application (typically 
+located in `main.c`).
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+#include <stdio.h>
+
+int main(void)
+{
+    puts("Hello World!");
+    return 0;
+}
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The application's Makefile                         {#the-applications-makefile}
+==========================
+The minimal Makefile                                    {#the-minimal-makefile}
+--------------------
+At minimum the Makefile of an application (see #the-build-system) needs the 
+following macros defined:
```

Comment:
Depending on the version of make [macro and variable are used interchangeably](http://www.gnu.org/software/make/manual/make.html#Using-Variables)